contentcuration/contentcuration/assets/channelList/vuex/channelList/actions.js

```diff
@@ -0,0 +1,114 @@
+import { ChannelListUrls, ChannelInvitationMapping } from '../../constants';
+import { isDummyId, getDirtyDiff } from '../../utils';
+import client from 'shared/client';
+
+/* CHANNEL LIST ACTIONS */
+export function loadChannelList(context, listType) {
+  return client.get(ChannelListUrls[listType]).then(response => {
+    const channels = response.data;
+    context.commit('ADD_CHANNELS', channels);
+    return channels;
+  });
+}
+
+export function loadAllChannels(context) {
+  return client.get(window.Urls['channelslim-list']()).then(response => {
+    const channels = response.data;
+    context.commit('ADD_CHANNELS', channels);
+    return channels;
+  });
+}
+
+export function addStar(context, channelId) {
+  context.commit('SET_CHANNEL_BOOKMARK', { id: channelId, bookmark: true });
```

Comment:
Do optimistic updating of the channel bookmark state here, to make the interface feel more responsive to the user.

In the case of failure to update, we rollback the update below.

Reply:
The interface will change - we probably should have some global way of reporting errors, but in a non-blocking and inobtrusive way.